Code/src/main/java/nl/utwente/group10/ui/components/blocks/input/GraphBlock.java

```diff
@@ -1,82 +1,87 @@
+import javafx.scene.layout.BorderPane;
 import javafx.scene.layout.Pane;
 import nl.utwente.group10.haskell.exceptions.HaskellException;
 import nl.utwente.group10.haskell.expr.Apply;
 import nl.utwente.group10.haskell.expr.Expr;
 import nl.utwente.group10.haskell.expr.Ident;
-import nl.utwente.group10.haskell.type.ConstT;
-import nl.utwente.group10.haskell.type.FuncT;
-import nl.utwente.group10.haskell.type.Type;
 import nl.utwente.group10.ui.CustomUIPane;
 import nl.utwente.group10.ui.components.anchors.InputAnchor;
+import nl.utwente.group10.ui.components.blocks.Block;
 
 import java.util.Iterator;
 import java.util.List;
+import java.util.Locale;
 import java.util.NoSuchElementException;
 
+/**
+ * Block that accepts a (Float -> Float) function to be displayed on a linechart
+ * inside the Block.
+ */
 public class GraphBlock extends Block implements InputBlock {
+    /** The InputAnchor of this Block. */
     private InputAnchor input;
 
-    @FXML
-    private Pane anchorSpace;
+    /** The Pane that contains the inputs. */
+    @FXML private Pane inputSpace;
 
-    @FXML
-    private LineChart<Double, Double> chart;
+    /** The LineChart that is displayed inside this Block. */
+    @FXML private LineChart<Double, Double> chart;
 
-    @FXML
-    private NumberAxis x;
+    /** NumberAxis for x. */
+    @FXML private NumberAxis x;
 
-    @FXML
-    private NumberAxis y;
+    /** NuberAxis for y. */
+    @FXML private NumberAxis y;
 
+    /**
+     * Constructs a new GraphBlock.
+     * @param pane The CustomUIPane on which this Block resides.
+     */
     public GraphBlock(CustomUIPane pane) {
         super(pane);
-
         loadFXML("GraphBlock");
 
-        input = new InputAnchor(this, pane);
-        anchorSpace.getChildren().setAll(input);
-    }
-
-    @Override
-    public Expr asExpr() {
-        return input.asExpr();
-    }
-
-    @Override
-    public Type getInputSignature(InputAnchor input) {
-        assert input.equals(this.input);
-        return getInputSignature(0);
-    }
-
-    @Override
-    public Type getInputSignature(int index) {
-        assert index == 0;
-        return new FuncT(new ConstT("Float"), new ConstT("Float"));
+        input = new InputAnchor(this);
+        input.layoutXProperty().bind(inputSpace.widthProperty().divide(2));
+        inputSpace.getChildren().setAll(input);
+        
+        //Make sure inputSpace is drawn on top.
+        BorderPane borderPane = (BorderPane) inputSpace.getParent();
+        borderPane.getChildren().remove(inputSpace);
+        borderPane.setTop(inputSpace);
     }
 
     @Override
     public List<InputAnchor> getAllInputs() {
         return ImmutableList.of(input);
     }
+    
+    @Override
+    public void updateExpr() {
+        this.expr = input.getExpr();
+        super.updateExpr();
+    }
 
     @Override
-    public void invalidateConnectionState() {
+    public void invalidateVisualState() {
+        super.invalidateVisualState();
```

Comment:
On the issue of blank methods: This is done to be able to have invalidateX be called before cascadeX.